### PR TITLE
ARROW-6091: [Rust] [DataFusion] Implement physical execution plan for LIMIT

### PR DESCRIPTION
**Pull request by andygrove**

### CODE REVIEW

File: rust/datafusion/src/execution/physical_plan/limit.rs [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    /// Maximum number of rows to return
+    limit: usize,
+}
+
+impl Partition for LimitPartition {
+    fn execute(&self) -> Result<Arc<Mutex<dyn BatchIterator>>> {
+        // collect up to "limit" rows on each partition
+        let threads: Vec<JoinHandle<Result<Vec<RecordBatch>>>> = self
+            .partitions
+            .iter()
```
Comment (alippai) [inline rust/datafusion/src/execution/physical_plan/limit.rs]: Would be an overkill to use rayon's par_iter()?

Comment (andygrove) [reply rust/datafusion/src/execution/physical_plan/limit.rs]: For this round my plan was to keep things simple and do a second round of optimizations that I've started tracking in https://jira.apache.org/jira/browse/ARROW-6689 and in particular see the subtask https://jira.apache.org/jira/browse/ARROW-6691 related to threading.

Comment (alippai) [reply rust/datafusion/src/execution/physical_plan/limit.rs]: I think that's a "no" for rayon :D

**Comment by andygrove** [comment]
@nevi-me @sunchao @paddyhoran Please review when you can. Thanks.

**Comment by andygrove** [comment]
@nevi-me @sunchao @paddyhoran Any objection to me merging this one?

**Review by paddyhoran — APPROVED** [review]
I only did a high level review but LGTM.  I'm in favor of merging and opening follow up issues if we find anything (particularly when it comes to your PR's to DataFusion) to keep the momentum going.

**Pull request closed without merging**